internal/bguploader.go

```diff
@@ -135,23 +158,48 @@ func (b *BgUploader) scanAndProcessFiles() {
 
 }
 
+// speculativelyGenerateNextWalFilenames continually generates and sends the
+// subsequent WAL segment names after b.initialWalSegment to fileChan until ctx
+// is canceled or 100 * b.maxNumUploaded filenames have been generated.
+func (b *BgUploader) speculativelyGenerateNextWalFilenames(ctx context.Context, fileChan chan<- string) {
+	walSegment := b.initialWalSegment
+	for i := 0; i < 100*int(b.maxNumUploaded); i++ {
```

Comment:
Why 100?

Reply:
This was arbitrary - I thought we should have _some_ limit here. That said, it is a bit paranoid. I'm fine with removing it - let me know what you think

Reply:
1. I cannot find limits of one upload anywhere else
2. It seems like here we should have infinite loop?
3. You still synchronously read directory in a please where result of chan passed to speculativelyGenerateNextWalFilenames() is used

I simply do not understand how it's intended to work